src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```diff
@@ -376,9 +376,9 @@ public function editCmsCategoryAction($cmsCategoryId, Request $request)
         $cmsPageCategoryFormBuilder = $this->getCmsPageCategoryFormBuilder();
         $cmsCategoryParentId = null;
 
+        $cmsPageCategoryForm = $cmsPageCategoryFormBuilder->getFormFor((int) $cmsCategoryId);
```

Comment:
It is on purpose that `$cmsPageCategoryFormBuilder->getFormFor()` is inside the `try`.

There is a hook being fired in `getFormFor()` so if a module plugs itself on it, it might throw an exception and break the whole page. We dont want that a merchant can break his page "just by installing a module" so we purposefully put all the functions firing hooks into big `try/catch` statements.

We must find another way to make phpstan happy.

Reply:
If I put all getFormFor in try. I need variables defined for the `$this->render` call.

Reply:
Then maybe we should put the `$this->render()` into the `try` too ?

Reply:
I'm not objective since I've always been against this exception wrapping 😅  I think the exception should thrown even if it's because of a module Or the merchant won't even understand that there is an error Just nothing happens..
Besides there will still be an error if the `$cmsPageCategoryForm` is not set because it's used by `render` so the page will still break anyway

Reply:
Ping  @PrestaShop/prestashop-core-developers

Reply:
Agreed with @jolelievre.

There is no reason to use `render` if there are missing variables, or if an exception is raised.

Reply:
After discussion, we need your advice @eternoendless

Reply:
Ping @matks #14193

Reply:
@matks @jolelievre @PierreRambaud Fixed after the meeting.